services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/BandwidthV2Spec.groovy

```diff
@@ -0,0 +1,253 @@
+package org.openkilda.functionaltests.spec.flows
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.functionaltests.extension.tags.Tag.SMOKE
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.tags.Tags
+import org.openkilda.functionaltests.helpers.PathHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.IslInfoData
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowState
+
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+
+@Narrative("Verify that ISL's bandwidth behaves consistently and does not allow any oversubscribtions etc.")
+class BandwidthV2Spec extends HealthCheckSpecification {
+
+    @Tags(SMOKE)
+    def "Available bandwidth on ISLs changes respectively when creating/updating/deleting a flow"() {
+        given: "Two active not neighboring switches"
+        def switchPair = topologyHelper.getNotNeighboringSwitchPair()
+
+        when: "Create a flow with a valid bandwidth"
+        def linksBeforeFlowCreate = northbound.getAllLinks()
+        def maximumBandwidth = 1000
+
+        def flow = flowHelper.randomFlow(switchPair)
```

Comment:
you can do `flowHelperV2.randomFlow` to get a V2 payload, so you won't have to convert it afterwards

Reply:
Updated.